### PR TITLE
update library dependency finder to use chain+

### PR DESCRIPTION
**Pull request by craiglink**
Set dependency finder to use [chain+](https://docs.platformio.org/en/latest/librarymanager/ldf.html) so that only needed libraries pulled in.

### CODE REVIEW

File: FluidNC/src/WebUI/TelnetServer.cpp [hunk 1]
```diff
@@ -1,12 +1,12 @@
 // Copyright (c) 2014 Luc Lebosse. All rights reserved.
 // Use of this source code is governed by a GPLv3 license that can be found in the LICENSE file.
-#include <ESPmDNS.h>
 #include "../Machine/MachineConfig.h"
 #include "TelnetClient.h"
 #include "TelnetServer.h"
 #include "WebSettings.h"
 
 #ifdef ENABLE_WIFI
+#include <ESPmDNS.h>
```
Comment (craiglink) [inline FluidNC/src/WebUI/TelnetServer.cpp:9]: ESPmDNS was being pulled in for bluetooth only and noradio builds

**Comment by MitchBradley** [comment]
Is there a problem with this?  It seems like a good idea in principle.

File: FluidNC/src/WebUI/TelnetServer.cpp (hunk 1 above)
Comment (craiglink) [reply FluidNC/src/WebUI/TelnetServer.cpp:9]: the change in this include makes sense

**Comment by craiglink** [comment]
This chain+ setting worked for a number of builds, but failed on a few additional ones after testing.  Specifically it wasn't finding wifi.h when compiling the custom websockets code.    not sure why so closed the MR until I can figure it out

**Pull request closed without merging**